code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -356,6 +356,10 @@ MATCHBOXES ARE ALSO IN FANCY.DM
 	smoketime--
 	var/datum/gas_mixture/env = location.return_air()
 	if(smoketime <= 0 | env.molar_density(GAS_OXYGEN) < (5 / CELL_VOLUME))
```

Comment:
🤔

Reply:
🤔

Reply:
Apparently the `|` has lower precedence than the comparisons for some fucking reason

Reply:
i am unable to decipher your message and thus my answer remains
🤔

Reply:
`|` `!=` `||`

Reply:
so replace the `|` with a `||`, got it

Reply:
In simple terms,
`(smoketime <= 0 | env.molar_density(GAS_OXYGEN)` first does "`smoketime <= 0`", returns 0 or 1, and then it bitwise OR (which is equivalent to regular OR as we have only 1 bit to go by

This is equivalent to what it is intended to do, but by pure mistake. Changing it to `||` will help readability and will look less stupid